packages/common/src/core/schemas/internal/getCategoryItems.ts

```diff
@@ -17,24 +18,40 @@ export async function getCategoryItems(
   const url = `${hubRequestOptions.portal}/portals/${orgId}/categorySchema`;
   try {
     const { categorySchema } = await request(url, hubRequestOptions);
-    return parseCategories(categorySchema[0].categories, []);
+    return convertCategoryProps(categorySchema[0].categories);
   } catch (e) {
     return [];
   }
 }
 
-function parseCategories(
-  categories: any[],
-  allCategories: any[]
+/**
+ * Convert the categories into a format that can be used
+ * by the IUiSchemaComboboxItem interface
+ *
+ * @param obj The object to convert
+ * @returns the converted object where `title` becomes `value` and `categories` becomes `children`
+ */
+function convertCategoryProps(
+  arrayOfCategories: any,
+  parentValue?: string
 ): IUiSchemaComboboxItem[] {
-  categories.forEach((c) => {
-    if (!c.categories?.length) {
-      allCategories.push({
-        value: c.title,
-      });
-    } else {
-      parseCategories(c.categories, allCategories);
-    }
+  return arrayOfCategories.map((category: any) => {
+    // if there is a parentValue, append the category title to it to give it a unique value
+    // that can be distinguished from other categories
+    const value = parentValue
+      ? `${parentValue} / ${category.title}`
+      : category.title;
+
+    // use just the title as the label, so we don't have redundant information visually
+    const label = category.title;
+
+    // if there are no more categories, return null for children, otherwise recursively call this function
+    return {
+      value: `${value}`,
+      label: `${label}`,
+      children: !category.categories?.length
+        ? null
```

Comment:
minor: should we default to an empty array rather than `null` if there are no children? Not sure if this has other implications

Reply:
As far as I'm aware from testing, it doesn't have any effect either way. I'll switch to an empty array for consistency.